### PR TITLE
Fix contract security

### LINKED ISSUE #24: Configure decimals
Adjust decimal multiplication to launch with a supply of 1,000,000,000,000 tokens in the constructor of `HMToken.sol`.

### LINKED ISSUE #25: Adjust Bulk Transfer Limit to newly adjusted token supply
See #24 and adjust the `BULK_MAX_VALUE`

### LINKED ISSUE #26: Rename Transferral event back to Transfer
Solhint linter through errors with `Transfer` event as there is a function named `transfer` as well. However the ERC20 standard dictates that the token should emit a `Transfer` event instead of our `Transferral event`.

Tasks:
- Rename `Transferral` back to `Transfer`
- Adjust `solhint` to discard these errors

### LINKED ISSUE #27: Revert burning tokens when calling `approve()` and `increaseApproval()`
Add the check `(_to == address(0)) return false;` to `approve()` and `increaseApproval()` functions

### PR DESCRIPTION
**Pull request by rbval**
This pull request fixes some errors in the contract security.
- Fix ERC20 compliance by using `Transfer` event
- Adjust totalSupply to decimals
- Adjust `BULK_MAX_VALUE` to totalSupply
- Adjust require statements so we don't burn tokens

Fixes #24
Fixes #25 
Fixes #26 
Fixes #27

### CODE REVIEW

**Comment by rbval** [comment]
Fixed tests, how does it look @posix4e ?

File: contracts/HMToken.sol [hunk 1]
```diff
@@ -12,47 +12,46 @@ contract HMToken is HMTokenInterface {
...
 
     constructor(uint256 _totalSupply, string _name, uint8 _decimals, string _symbol) public {
-        totalSupply = _totalSupply;
+        totalSupply = _totalSupply * (10 ** uint256(_decimals));
         name = _name;
         decimals = _decimals;
         symbol = _symbol;
-        balances[msg.sender] = _totalSupply;
+        balances[msg.sender] = totalSupply;
+        BULK_MAX_VALUE = totalSupply;
```
Comment (gaieges) [inline contracts/HMToken.sol]: A constant is assignable on construction?

Comment (rbval) [reply contracts/HMToken.sol]: Wow what a blunder, will fix that

Comment (gaieges) [reply contracts/HMToken.sol]: I honestly dont know, was just asking if its even possible

**Pull request merged**